TickSpec/ServiceProvider.fs

```diff
@@ -3,21 +3,94 @@
 open System
 open System.Collections.Generic
 open System.Diagnostics
+open System.Reflection
 
+/// Provides an instance provider for tests
+type IInstanceProvider =
+    inherit IDisposable

```

Comment:
in general people don't do this (and/or it causes a mess when they do) - `IDisposable` is an orthogonal interface and we should not force people to implement it